projects/common/src/utilities/formatters/date/date-formatter.ts

```diff
@@ -58,25 +62,27 @@ export class DateFormatter {
     this.options = this.applyOptionDefaults(options);
   }
 
-  public format(value: Date | number | undefined | string): string {
-    return this.convertDateToString(value);
+  public format(value: Date | number | undefined | string, timezone?: string): string {
```

Comment:
I thought a date already encodes a time zone? If I'm mistaken, this makes sense. If it does, we shouldn't accept a different one since that's already expressible and leads to ambiguity

Reply:
Yeah, `Date` does have timezone information, the issue though is that the formatter takes a union that may or may not be a `Date` so the extra param would _sometimes_ be necessary.

Maybe I should name it `timezoneOverride`? Not sure; open to suggestions.

Reply:
My 2p: the caller has the option of using a date object if they wish to include time zone information, the string and number versions are just conveniences and wouldn't try to make the api support using them with a custom time zone.

Reply:
Okay, so played with this a bit more and while you can apply timezone info to `Date`, when that gets formatted it it just adjusts the time to zulu time. You can see this in the matching test. If our `Date` is built with timezone info like `2021-08-19T14:23:56.789-08:30`, once we run it through the formatter without a `timezone` arg, it spits out the zulu time equivalent of `22:53:56Z` and the original timezone is lost.

Keeping the `timezone` argument is actually additional information that tells the formatter to spit out the time for that timezone. So in the same example passing `2021-08-19T14:23:56.789-08:30` with `timezone` set to `PST` would provide us the expected `14:53:56-08:00`.